syntax = "proto3";

package grpc_base;

//
// Service definitions
//

service Netatmo {
    rpc GetData(NetatmoMessages.NetatmoRequest) returns (NetatmoMessages.NetatmoData) {}; 
}

service Image {
    rpc GetImage(ImageMessages.ImageRequest) returns (ImageMessages.ImageData) {};
}

//
// Message definitions
//

message NetatmoMessages {
    message NetatmoRequest {}

    message NetatmoData {
        IndoorData indoor = 1;
        OutdoorData outdoor = 2;
    }

    message IndoorData {
        float temperature = 1;
        float CO2 = 2;
        float humidity = 3;
        float noise = 4;
        float pressure = 5;
    }

    message OutdoorData {
        float temperature = 1;
        float humidity = 2;
    }
};

message ImageMessages {
    message ImageRequest {
        string imagePath = 1;
        int32 width = 2;
        int32 height = 3;
    }

    message ImageData {
        bytes data = 1;
        string format = 2;
        int32 width = 3;
        int32 height = 4;
    }
}